src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/MessageParser.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+
+namespace Microsoft.AspNetCore.Sockets.Internal.Formatters
+{
+    public class MessageParser
+    {
+        private TextMessageParser _textParser = new TextMessageParser();
+        private BinaryMessageParser _binaryParser = new BinaryMessageParser();
+
+        public void Reset()
+        {
+            _textParser.Reset();
+            _binaryParser.Reset();
+        }
+
+        public bool TryParseMessage(ref BytesReader buffer, MessageFormat format, out Message message)
+        {
+            return format == MessageFormat.Text ?
+                _textParser.TryParseMessage(ref buffer, out message) :
+                _binaryParser.TryParseMessage(ref buffer, out message);
+        }
+
+        public static MessageFormat GetFormat(byte formatIndicator)
```

Comment:
No, this is a method required only during parsing, it turns the format indicator byte into the `MessageFormat` enum. I could consolidate them but these are public APIs that will be used by consumers so I don't know that they fit on a "Utils" class really.